File: solutions/haskell/050-climbing-stairs.hs
{-
  Problem 50: Climbing Stairs (LeetCode 70)
  Difficulty: Easy
  Language: Haskell
-}
climbStairs :: Int -> Int
climbStairs n = fst $ foldl (\(a, b) _ -> (b, a + b)) (1, 1) [1..n]

File: solutions/scala/050-climbing-stairs.scala
/**
 * Problem 50: Climbing Stairs (LeetCode 70)
 * Difficulty: Easy
 * Language: Scala
 */
def climbStairs(n: Int): Int =
  (1 until n).foldLeft((1, 1)) { case ((a, b), _) => (b, a + b) }._1
